### ISSUE TITLE
Optional TrimSpace

### ISSUE THREAD

**Issue by wallaceicy06** [ianlopshire/go-fixedwidth#34]
The decoder automatically trims spaces which I guess works for most people. I have a case where I do not want to trim spaces. Would it be possible to allow this to be configured via an option on the decoder?

**Comment by wallaceicy06** [comment]
Duplicate of #35

**Comment by ianlopshire** [comment]
@wallaceicy06 does #28/#37 solve this use-case for you?

After #37 is merged the decoder will respect alignment and padding character when decoding the file. You can look at [decode_test.go#L166](https://github.com/ianlopshire/go-fixedwidth/blob/feature/configurable-formatting/decode_test.go#L166) as an example.

**Comment by wallaceicy06** [comment]
Yeah, that looks like a better more generic solution. I'll watch that issue and drop my PR.
